src/trip_kinematics/URDFParser.py

```diff
@@ -0,0 +1,301 @@
+import xml.etree.ElementTree as ET
+from collections import defaultdict
+from typing import Dict, List
+
+import numpy as np
+from scipy.spatial.transform import Rotation as ScipyRotation

```

Comment:
scipy is not a requirement!
If you want to calculate quaternions from rotation matrices we should further discuss wheter to directly add the functionality to TriP or use a outside library